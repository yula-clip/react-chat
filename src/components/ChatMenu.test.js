/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ChatMenu from "./ChatMenu";

const mockProps = {
  activeUser: {
    isMember: true,
    isCreator: true,
    isChatMember: true
  },
  disabled: false,
  onLeaveClick: jest.fn(),
  onDeleteClick: jest.fn()
};

describe("<ChatMenu />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ChatMenu {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<ChatMenu {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
