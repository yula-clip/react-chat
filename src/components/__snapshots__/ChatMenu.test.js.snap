// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMenu /> renders correctly 1`] = `
<button
  aria-haspopup="true"
  aria-owns={null}
  className="MuiButtonBase-root-36 MuiIconButton-root-30 MuiIconButton-colorInherit-31"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-35"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-39"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-52"
  />
</button>
`;
